spinnman/spalloc/abstract_classes.py

```diff
@@ -0,0 +1,716 @@
+# Copyright (c) 2022 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+API of the client for the Spalloc web service.
+"""
+
+from sqlite3 import Cursor
+import struct
+import time
+from typing import Callable, Dict, Iterable, Set, Tuple
+from spinn_utilities.abstract_base import (
+    AbstractBase, abstractmethod, abstractproperty)
+from spinn_utilities.overrides import overrides
+from spinn_utilities.abstract_context_manager import AbstractContextManager
+from spinnman.connections.abstract_classes import Listenable
+from spinnman.connections.udp_packet_connections import (
+    update_sdp_header_for_udp_send, SCAMPConnection, EIEIOConnection,
+    BootConnection)
+from spinnman.connections.udp_packet_connections.boot_connection import (
+    _ANTI_FLOOD_DELAY)
+from spinnman.constants import SCP_SCAMP_PORT
+from spinnman.exceptions import SpinnmanTimeoutException
+from spinnman.messages.eieio import (
+    AbstractEIEIOMessage,
+    read_eieio_command_message, read_eieio_data_message)
+from spinnman.messages.sdp import SDPMessage, SDPFlag, SDPHeader
+from spinnman.messages.scp.abstract_messages import AbstractSCPRequest
+from spinnman.messages.scp.impl import IPTagSet
+from spinnman.messages.scp.enums import SCPResult
+from spinnman.messages.spinnaker_boot import SpinnakerBootMessage
+from spinnman.transceiver import Transceiver
+from .enums import SpallocState
+
+_ONE_SHORT = struct.Struct("<H")
+_TWO_SHORTS = struct.Struct("<2H")
+_TWO_SKIP: bytes = b'\0\0'
+_NUM_UPDATE_TAG_TRIES = 3
+_UPDATE_TAG_TIMEOUT = 1.0
+
+
+class AbstractSpallocClient(object, metaclass=AbstractBase):
```

Comment:
I would suggest that this and the other classes in this file are broken out into files of their own.  This makes them easier to find in the future...

Reply:
Done